datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -255,39 +256,61 @@ where
             if start > to {
                 return None;
             }
-            let end = start + step;
-            debug!(logger, "Requesting logs for blocks [{}, {}]", start, end);
+
+            // This will iterate more than once iff `step` has been reduced,
+            // making as many parallel requests of size `step` as necessary,
+            // respecting `LOG_STREAM_PARALLEL_CHUNKS`.
+            let mut chunk_futures = vec![];
+            let mut offset = start;
+            for _ in 0..*LOG_STREAM_PARALLEL_CHUNKS {
+                if offset == to + 1 {
+                    break;
+                }
+                let log_filter = log_filter.clone();
+                let next_offset = (offset + step).min(to) + 1;
```

Comment:
Could get rid of this `+1` and the `-1` below if you just adding the `1` when you set the `offset` below.

Reply:
That will be much better, thanks!